src/simulators.jl

```diff
@@ -202,14 +217,27 @@ function simulate!(sys,
     neighbors = find_neighbors(sys, sys.neighbor_finder; n_threads=n_threads)
     run_loggers!(sys, neighbors, 0, run_loggers; n_threads=n_threads)
 
+    using_constraints = (length(sys.constraints) > 0)
```

Comment:
This can be used in all simulators for clarity rather than always checking `length(sys.constraints) > 0`.